testsupport/space.go

```diff
@@ -117,6 +120,38 @@ func getSpaceTargetMember(t *testing.T, awaitilities wait.Awaitilities, space *t
 	return nil
 }
 
+// func VerifySpaceRelatedResources(t *testing.T, awaitilities wait.Awaitilities, userSignup *toolchainv1alpha1.UserSignup, tierName string) {
+// TODO: merge with `VerifyResourcesProvisionedForSpace()` func below?
+func VerifySpaceRelatedResources(t *testing.T, awaitilities wait.Awaitilities, spaceName, tierName, creatorName string) *toolchainv1alpha1.Space {
+	hostAwait := awaitilities.Host()
+
+	tier, err := hostAwait.WaitForNSTemplateTier(tierName)
+	require.NoError(t, err)
+	hash, err := testtier.ComputeTemplateRefsHash(tier) // we can assume the JSON marshalling will always work
+	require.NoError(t, err)
+
+	space, err := hostAwait.WaitForSpace(spaceName,
+		wait.UntilSpaceHasTier(tierName),
+		wait.UntilSpaceHasLabelWithValue(toolchainv1alpha1.SpaceCreatorLabelKey, creatorName),
+		wait.UntilSpaceHasLabelWithValue(fmt.Sprintf("toolchain.dev.openshift.com/%s-tier-hash", tierName), hash),
+		wait.UntilSpaceHasStateLabel(toolchainv1alpha1.SpaceStateLabelValueClusterAssigned),
+		wait.UntilSpaceHasConditions(Provisioned()),
+		// wait.UntilSpaceHasStatusTargetCluster(mur.Spec.UserAccounts[0].TargetCluster),
+	)
+	require.NoError(t, err)
+
+	memberAwait, err := awaitilities.Member(space.Spec.TargetCluster)
+	require.NoError(t, err)
+
+	// Verify provisioned NSTemplateSet
+	nsTemplateSet, err := memberAwait.WaitForNSTmplSet(space.Name, wait.UntilNSTemplateSetHasTier(tierName))
+	require.NoError(t, err)
+	tierChecks, err := tiers.NewChecksForTier(tier)
+	require.NoError(t, err)
+	tiers.VerifyNSTemplateSet(t, hostAwait, memberAwait, nsTemplateSet, tierChecks)
+	return space
+}
```

Comment:
it's not a new function, I simply moved it: https://github.com/codeready-toolchain/toolchain-e2e/pull/518/files#diff-290814afc5de2e13eaecfa10f4623ffd38b7947e1db26da25b3e5fdfedf8f34eL140
And I agree with you, there is a big overlap, but I suggest we resolve this in a subsequent PR

Reply:
but it's not only a copy-paste action - there are other changes in the content of the function, right? 
* You removed some stuff
* commented out the line 139
* replaced `GetMurTargetMember(t, awaitilities, mur)` with `awaitilities.Member(space.Spec.TargetCluster)`

which means that this function needs to be reviewed and modified again anyway.
Reusing the existing functions would make the changes of the PR smaller, so it would be beneficial. 

If you insist on removing this function in the following PR, then I'm fine, but please don't make it big - just reuse the already existing code or slightly adjust it.

Reply:
I've reverted this change in d772e42. 
We can take care about it later